### PR TITLE
documentation for packaging support (REVIEW)

### PR DESCRIPTION
**Pull request by rjeschmi**
Just opening this to get some advice on structure before I go too far :)

### CODE REVIEW

**Comment by rjeschmi** [comment]
The preview can be had here:

https://rjeschmi-eb-draft.readthedocs.org/en/develop/

File: docs/Creating_binary_packages.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+.. _creating_binary_pages:
+
+Creating Binary Packages
```
Comment (boegel) [inline docs/Creating_binary_packages.rst]: should be `Creating binary packages` (to be consistent with other page titles)

File: docs/Creating_binary_packages.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+========================
+
+As an experimental feature, support for `FPM <https://github.com/jordansissel/fpm>`_ has been integrated into EasyBuild. This creates a dependency on Ruby and a few other RubyGems and they have been provided through `Ruby-2.1.5.eb <https://github.com/rjeschmi/easybuild-easyconfigs/blob/a297788484eb778faeef5c9f335a58cc3b33c9b4/easybuild/easyconfigs/r/Ruby/Ruby-2.1.5.eb>`_.
```
Comment (boegel) [inline docs/Creating_binary_packages.rst]: make it clear what 'experimental' means

Comment (boegel) [inline docs/Creating_binary_packages.rst]: mention which EB version includes (experimental) support

File: docs/Creating_binary_packages.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+
+First you will need the Ruby package that includes FPM::
+
+    [easybuild@b018c9b7ec95 ~]$ eb Ruby-2.1.5.eb
```
Comment (boegel) [inline docs/Creating_binary_packages.rst]: drop the part before the `$`

File: docs/Creating_binary_packages.rst [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    == COMPLETED: Installation ended successfully
+    == 2015-05-18 15:12:58,526 main.easyblock INFO Results of the build can be found in the log file /export/easybuild/software/Ruby/2.1.5/easybuild/easybuild-Ruby-2.1.5-20150518.151258.log
```
Comment (boegel) [inline docs/Creating_binary_packages.rst]: use `/home/example` instead of `/export/easybuild`

File: docs/Creating_binary_packages.rst [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    == Results of the build can be found in the log file /export/easybuild/software/Ruby/2.1.5/easybuild/easybuild-Ruby-2.1.5-20150518.151258.log
```
Comment (boegel) [inline docs/Creating_binary_packages.rst]: `/home/example`

File: docs/Creating_binary_packages.rst [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+Creating Binary Packages
+========================
+
+As an experimental feature, support for `FPM <https://github.com/jordansissel/fpm>`_ has been integrated into EasyBuild. This creates a dependency on Ruby and a few other RubyGems and they have been provided through `Ruby-2.1.5.eb <https://github.com/rjeschmi/easybuild-easyconfigs/blob/a297788484eb778faeef5c9f335a58cc3b33c9b4/easybuild/easyconfigs/r/Ruby/Ruby-2.1.5.eb>`_.
+
+
+First you will need the Ruby package that includes FPM::
+
+    [easybuild@b018c9b7ec95 ~]$ eb Ruby-2.1.5.eb
+    == COMPLETED: Installation ended successfully
```
Comment (boegel) [inline docs/Creating_binary_packages.rst]: add `...` where you cut things out

**Comment by boegel** [comment]
@rjeschmi: some minor style-related remarks, thanks for looking into this!

One thing to keep in mind is to organise the page in different sections, like:

* `Preliminary requirements` (Ruby+FPM, e.g. via a module provided by EB, or as a system-wide manual installation (`gem install fpm`?);
* `Scope`: creating packages for software (and module) installed by EB, one package file per software application/tools, etc.
* `Usage`: which command line options, how to configure, etc.
* `Example: creating packages for HDF5 and dependencies` (or any other example, but something that includes dependencies next to the toolchain)

**Comment by rjeschmi** [comment]
thanks for looking at this @boegel. It give me some direction before I get too far into things :)

I hope we can use this as the test docs as well (pointing people at my branch before we merge the feature into experimental), so some of the wording might be a bit imprecise until we have a particular target merge.

With everything sitting in one file it should be simple to merge whenever we are ready.

**Comment by boegel** [comment]
Agreed, love the effort you're putting into this!

File: docs/Creating_binary_packages.rst (hunk 2 above)
Comment (rjeschmi) [inline docs/Creating_binary_packages.rst]: Maybe I'll outline the mainline documentation for experimental under Advanced and reference that?

**Comment by boegel** [comment]
Going in, thanks @rjeschmi!

**Pull request merged**